src/file_parsing.jl

```diff
@@ -60,6 +46,18 @@ struct AliasParamDict{FT} <: AbstractParamDict{FT}
     data::Dict
     "either a nothing, or a dictionary representing an override parameter TOML file"
     override_dict::Union{Nothing, Dict}
+    "Alias->name map"
+    alias_to_name_map::Dict
+end
+
+function AliasParamDict{FT}(
+    data::Dict,
```

Comment:
are the data entries indexed by names or aliases, could we specify this in a constructor docstring? And I assume the override_dict has the same structure?

Reply:
This is just an outer constructor that determines the `alias_to_name_map` (perhaps it should be an inner constructor, though). I believe that the data entries are always indexed by the names (correct me if I'm wrong, @odunbar).

Maybe @odunbar can suggest a better doc string? I believe, or would assume, that the override dict is restricted to a subset of the input dict?

Reply:
@ilopezgp, since this PR is not changing the existing interface or user-facing behavior, let's open this as an issue if it's about the name of the type.

Reply:
I'm actually planning on splitting this part out since it's not really related to removing old method support.